### PR TITLE
Feat/user page

### PR DESCRIPTION
**Pull request by GAMZAMANDU**
## 💡 개요
1. user-page의 Tab 간 섹션 전환을 완성했습니다.
2. posts를 제외한 user-page창의 퍼블리싱을 완료했습니다.

## 📃 작업내용
- tab이 home일때 화면 제작.
- tab이 projects일때 화면 (전체, 개인, 협업)으로 나뉨.

## 🔀 변경사항
- user-pages의 변경부분을 컴포넌트화했습니다.

## 📸 스크린샷
스크린샷 2024-11-25 오후 10 16 31
스크린샷 2024-11-25 오후 10 16 46

### CODE REVIEW

**Review by jaehwan02 — CHANGES_REQUESTED** [review]
### 프로젝트 상하 gap이 너무 좁아요
---
image

### user 접속시 생기는 오류 (useEffect 미사용)
---
image

### next/head가 아닌 Metadata API를 사용할 필요가 있음
---
image
image

File: src/app/components/detail/DetailBox.tsx [hunk 1]
```diff
@@ -15,7 +15,7 @@ const symbolName = {
 
 const DetailBox: React.FC<DetailBoxProps> = ({ symbol, length, data }) => {
 	return (
-		<div className="block w-[25rem] overflow-auto min-h-[9rem] h-auto p-14px mb-[1.2rem] border border-strokeColor rounded-lg">
+		<div className="block w-[21.0625rem] overflow-auto min-h-[9rem] h-auto p-14px mb-[1.2rem] border border-strokeColor rounded-lg">
```
Comment (obtuse-triangle) [inline src/app/components/detail/DetailBox.tsx:18]: min-w, max-w와 w-[30%]로 반응형 설계하기 [태블릿 등 데스크톱보다 작은 너비]

File: src/app/components/ProjectItem.tsx [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  tag,
+  category,
+  title,
+  description,
+}) => {
+  const imageUrl = url || defaultImg;
+
+  return (
+    <div className="w-fit h-fit">
+      <div className="w-[21.0625rem] h-[11.8125rem] bg-[#F1F1F1] rounded-2xl overflow-hidden">
```
Comment (obtuse-triangle) [inline src/app/components/ProjectItem.tsx:24]: min-w, max-w와 w-[30%]로 반응형 설계하기 [태블릿 등 데스크톱보다 작은 너비] 22
height 소수점 절삭하기

File: src/app/components/section/UserHome.tsx [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+import ProjectItem from '@components/ProjectItem';
+
+const UserHome = () => {
+	return (
+		<div>
+      <div className="w-full max-h-32 text-titleColor mt-4">마크다운 입력하는 곳</div>
+			<DetailBoxes />
+			<h3 className="text-titleColor font-bold text-2xl mt-2 mb-2">개인 프로젝트</h3>
+			<div className="flex gap-[0.7rem] flex-wrap">
+				<ProjectItem tag="개발완료" category="Desktop Utility" title="SANDDOET-App" description="산뜻 - AI 추천 RSS로 정보 습득이 쉬워지는 순간!" />
```
Comment (obtuse-triangle) [inline src/app/components/section/UserHome.tsx:12]: interface 타입 만들어서 map으로 출력하게 하기

File: tsconfig.json [hunk 4]
```diff
@@ -20,7 +20,8 @@
     ],
     "paths": {
       "@/*": ["./src/*"],
-      "@components/*": ["./src/app/components/*"]
+      "@components/*": ["./src/app/components/*"],
+      "@public/*": ["./public/*"],
```
Comment (obtuse-triangle) [inline tsconfig.json:24]: 👍

**Review by obtuse-triangle — CHANGES_REQUESTED** [review]

**Comment by obtuse-triangle** [comment]
This pull request introduces several new components and updates to the `src/app/components` directory to enhance the user interface and functionality. The changes include the addition of new components for displaying project items, user home sections, user posts, and user projects, as well as updates to existing components to improve layout and responsiveness.

New components:

* [`src/app/components/ProjectItem.tsx`](diffhunk://#diff-08de2d869f4ef04fe67ca9e11fc7fa44668d56ea879241f22da79f0c77173a37R1-R46): Added a new `ProjectItem` component to display individual project details, including an image, tag, category, title, and description.
* [`src/app/components/section/UserHome.tsx`](diffhunk://#diff-6279e34a16b01996b716e47bd52b7085486f4012ca1335fce320e5004d0a11e4R1-R21): Created a new `UserHome` component that includes a section for markdown input, detail boxes, and a list of personal projects using the `ProjectItem` component.
* [`src/app/components/section/UserPosts.tsx`](diffhunk://#diff-521b791d62556ddb102c2c3871e779b2fc0ce8ff1bcd5f64aac98f2a044c0449R1-R9): Added a new `UserPosts` component as a placeholder for user posts.
* [`src/app/components/section/UserProjects.tsx`](diffhunk://#diff-3c03749696f5467b6b6a07385e5b2f79eef9015c8c47cf6e429aa0a9d5f8048cR1-R31): Introduced a new `UserProjects` component with tabs for different project categories and a list of project items.
* [`src/app/components/section/UserTabs.tsx`](diffhunk://#diff-39700455215d7ee932660e3d8aae37706a7007b998610cbdb8c7e60d60df54a4R1-R26): Implemented a new `UserTabs` component to handle tab navigation between home, projects, and posts sections using URL parameters.

Updates to existing components:

* [`src/app/components/detail/DetailBox.tsx`](diffhunk://#diff-39224e0e17e7a411daa6b4d9cd84c4633e44117caeb1891189ae83708c39fb13L18-R18): Adjusted the width of the `DetailBox` component to improve layout consistency.
* [`src/app/components/detail/DetailBoxes.tsx`](diffhunk://#diff-39f6d94b035515bb22f2cf889b224f7e7ac9ab6e06ca6858b774e84050a60c25L7-R7): Modified the `DetailBoxes` component to use a flexible layout with wrapping and adjusted gaps between items.
* [`src/app/user/template.tsx`](diffhunk://#diff-83594f342ec877fc4aba22f0a069c283a5be81748fbd140ef5cfe763a3437a49L4-R4): Updated the `UserTemplate` component to replace the markdown section and detail boxes with the new `UserTabs` component for improved navigation and organization. [[1]](diffhunk://#diff-83594f342ec877fc4aba22f0a069c283a5be81748fbd140ef5cfe763a3437a49L4-R4) [[2]](diffhunk://#diff-83594f342ec877fc4aba22f0a069c283a5be81748fbd140ef5cfe763a3437a49L32-R34)

**Comment by obtuse-triangle** [comment]
// TODO: Detail.tsx의 'use client'제거

**Review by jaehwan02 — APPROVED** [review]

**Review by obtuse-triangle — APPROVED** [review]

File: src/app/components/detail/DetailBox.tsx (hunk 1 above)
Comment (obtuse-triangle) [reply src/app/components/detail/DetailBox.tsx:18]: #5 로 이동

File: src/app/components/ProjectItem.tsx (hunk 2 above)
Comment (obtuse-triangle) [reply src/app/components/ProjectItem.tsx:23]: #5 로 이동

**Pull request merged**